### ISSUE TITLE
[Bug Report] VAutocomplete is filtering out non text item

### ISSUE THREAD

**Issue by Glandos** [vuetifyjs/vuetify#11451]
### Environment
**Vuetify Version:** 2.3.0-beta.6
**Last working version:** 2.2.28
**Vue Version:** 2.6.11
**Browsers:** Firefox 76.0
**OS:** Linux x86_64

### Steps to reproduce
My VAutocomplete instance contains object items. By using `filter` function as stated in the documentation, it receives `item: object`, so I can filter the object with my own rules, and return either `true` or `false`.

### Expected Behavior
When `filter` function returns `true` for an item, it must be shown.

### Actual Behavior
It is filtered out by f8068ed6d4905c953463cf898150aae0b940fc3e for no explicit reason.

### Reproduction Link
<a href="https://codepen.io/glandos/pen/LYpMdGO" target="_blank">https://codepen.io/glandos/pen/LYpMdGO</a>

### Other comments
Maybe update the documentation to requires that `getText` must return a value? But I see no reason to remove items if the user wants to filter it manually.

**Comment by Glandos** [comment]
By the way, it also remove dividers, which seems to break the menu layout.

**Comment by johnleider** [comment]
This was a change made for #11415

**Comment by johnleider** [comment]
The dividers and headers were never shown when searching; I'm not sure how this affects the layout. Both my fix and the proposal in #11442 made assumptions on what the user would want. #11442 checks to see if the object is specifically `divider` or `header` while f8068ed says if there isn't a text value to display, there is no point in sending back the result.

With that being said, I agree that the items should pass to the filter regardless. This is what I'm planning on implementing as a fix that keeps #11415 resolved as well as resolving this one.

```ts
{
  filteredItems (): object[] {
    if (!this.isSearching || this.noFilter || this.internalSearch == null) return this.allItems

    return this.allItems.filter(item => {
      const value = getPropertyFromItem(item, this.itemText, undefined)
      const text = value ? String(value) : ''

      return this.filter(item, String(this.internalSearch), text)
    })
  }
}
```

**Comment by Glandos** [comment]
Your correction is OK for me, many thanks for that!

I have to admit that I was in a hurry when submitting the bug, and I didn't really explain why I needed this. I am using `VAutocomplete` in an unusual way that can display multiple elements depending on what's typed, and separate elements by dividers if needed. This is like a search box with different sources and different result types. So I still need header and dividers after filtering, based on my custom `filter` function.
